src/neuroconv/nwbconverter.py

```diff
@@ -39,8 +40,7 @@ def get_source_schema(cls):
             source_schema["properties"].update({interface_name: unroot_schema(data_interface.get_source_schema())})
         return source_schema
 
-    @classmethod
-    def get_conversion_options_schema(cls):
+    def get_conversion_options_schema(self):
```

Comment:
Is changing this from a classmethod really necessary? I'd rather not do that

Reply:
The problem is making line 52 reference the instance attribute instead of the class one. Maybe there is another way? I will think about it.

Another option is to set the class attribute data_interface_classes from inside the init . Is there a standard way to do that?

Reply:
why can't you just do `cls.data_interface_classes = ...`

Reply:
We discussed this in our meeting, this is easily fixable via a small `__new__` override to allow communication between these attributes for the instance of the class, very similar to how the `BaseExtractorInterface` works.

I also pointed out that an alternative approach here would be to have a separate but related `NWBConverter` class (possibly `DynamicNWBConverter`) that takes the instantiated list of data interfaces as input and adjusts any of the methods as required by the adjusted workflow, not dissimilar to this old PR of a utility-like function for generating such a class from a list of data interface classes: https://github.com/catalystneuro/nwb-conversion-tools/pull/110/files

Reply:
@bendichter It can be done. Check the last commit as I have reverted this. 

My concern with updating class attributes is that if you initialize another `NWBConverter` instance the updates of one will propagate to the other entangling them in hard to predict (and debug) ways.

```
interface1 = NWBConverter()  # This modifies the class attributes
interface2 = NWBConverter()  # This modifies the class attributes as well

interface1.run_conversion() #  <- here the interface1 will have the class attributes as set by interface2 
```
If this is an exotic concern that you think is unlikely to affect us then we can keep the code as it is (check the last commit). If on the other hand this is a concern after thinking more about it I think that I favor the alternative proposed by @CodyCBakerPhD of another class that works as a NWBConverter child factory.  We can try to emulate the [similar process in scikit-learn](https://scikit-learn.org/stable/modules/compose.html) as much as possible to leverage their data model as well 

What do you think?

Reply:
I really want to avoid using meta programming as I think is harder to do introspection and therefore debug. A simple way to:
1) Keep the code `NWBConverter` without breaking backwards compatibility.
2) Provide the functionality disused in this PR
3) Not using metaprogramming .

Is to introduce use something like the following class instead:

```python
class InterfacePipeline(NWBConverter):

    def get_conversion_options_schema(self):
        """Compile conversion option schemas from each of the data interface classes."""
        conversion_options_schema = get_base_schema(
            root=True,
            id_="conversion_options.schema.json",
            title="Conversion options schema",
            description="Schema for the conversion options",
            version="0.1.0",
        )
        for interface_name, data_interface in self.data_interface_classes.items():
            conversion_options_schema["properties"].update(
                {interface_name: unroot_schema(data_interface.get_conversion_options_schema())}
            )
        return conversion_options_schema


    def __init__(self, data_interfaces, verbose=True):
        self.verbose = verbose
        if isinstance(data_interfaces, list):
            # Create unique names for each interface
            counter = {interface.__class__.__name__: 0 for interface in data_interfaces}
            total_counts = Counter([interface.__class__.__name__ for interface in data_interfaces])
            self.data_interface_objects = dict()
            for interface in data_interfaces:
                class_name = interface.__class__.__name__
                counter[class_name] += 1
                unique_signature = f"{counter[class_name]:03}" if total_counts[class_name] > 1 else ""
                interface_name = f"{class_name}{unique_signature}"
                self.data_interface_objects[interface_name] = interface
        elif isinstance(data_interfaces, dict):
            self.data_interface_objects = data_interfaces
            
            
        self.data_interface_classes = {
                name: interface.__class__ for name, interface in self.data_interface_objects.items()
            }
```

I tested this locally and it works. Critically, we need to make the class method `get_conversion_options_schema` and instance method here as the class method relies on the class attribute. The other private method  to validate the source schema is not required since the instances of the interfaces are already initialized. 

You can find a complete script with this here:
https://gist.github.com/h-mayorquin/72b83aaa9c19375862819651e64000c5

Reply:
Ok, I implemented this in #169.